task_5/Oleg.Mikhailov/src/main/java/task6/Binary.java

```diff
@@ -1,20 +1,22 @@
 package task6;
 
+import static jdk.nashorn.internal.objects.NativeArray.reverse;
+
 public class Binary {
     public static void main(String[] args) {
         int i = 0;
         int number =30;
-        int[] mas = new int[]{0,0,0,0,0};
+
+        String numb = "";
         while(i < 5) {
 
             int tmp = number / 2;
-            int bin = number - tmp*2;
+            int bin = number % 2;
             number=tmp;
-            mas[i] = bin;
+            numb = numb + bin;
```

Comment:
Every time you concatenate a String, Java creates a new String object. Use StringJoiner class instead of String for the "numb" variable and add new "bin" by calling its append method.